packages/components/src/utils/useExitingAnimation/useExitingAnimation.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react'
+
+/** a utility hook for handling an the addition and removal of an animation and exit animation class */
+export const useExitingAnimation = (
+  shouldAnimate: boolean | undefined,
+  animatingClass: string,
+  exitingClass: string,
+  exitDuration: number = 1000,
+): string | undefined => {
+  const [animationClass, setAnimationClass] = useState('')
+  const prevIsAiLoading = React.useRef<boolean | undefined>(undefined)
```

Comment:
It looks like this hook could be used in different components, so we should probably rename this so it's not specific to ai moment

```suggestion
  const prevIsLoading = React.useRef<boolean | undefined>(undefined)
```